app/CentreUser.php

```diff
@@ -129,14 +130,26 @@ protected function homeCentres()
      *
      * @return Collection
      */
-    public function relevantCentres()
+    public function relevantCentres($programme = 0)
     {
         // default to empty collection
         $centres = collect([]);
         switch ($this->role) {
             case "foodmatters_user":
-                // Just get all centres
-                $centres = collect(Centre::get()->all());
+                if ($programme) {
+                    $centres = DB::table('centres')
+                        ->leftJoin('sponsors', 'centres.sponsor_id', 'sponsors.id')
```

Comment:
this could just be 
```
                        ->leftJoin('sponsors', 'centres.sponsor_id', 'sponsors.id')
                        ->where('sponsors.programme', $programme)
                        ->get()



```